File: leetcode-04/solutions/cpp/0945.cpp
class Solution {
 public:
  int minIncrementForUnique(vector<int>& A) {
    int ans = 0;
    int min = 0;

    sort(A.begin(), A.end());

    for (int a : A) {
      ans += max(min - a, 0);
      min = max(min, a) + 1;
    }

    return ans;
  }
};

File: leetcode-04/solutions/python3/0945.py
class Solution:
    def minIncrementForUnique(self, A: List[int]) -> int:
        ans = 0
        min = 0
        
        A.sort()
        
        for a in A:
            ans += max(min - a, 0)
            min = max(min, a) + 1
        
        return ans